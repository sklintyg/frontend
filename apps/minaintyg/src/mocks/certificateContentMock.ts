export const certificateContentMock = [
  {
    heading: 'Smittbärarpenning',
    body: `
      <h3>Avstängning enligt smittskyddslagen på grund av smitta</h3>
      <p>Ej angivet</p>
    `,
  },
  {
    heading: 'Grund för medicinskt underlag',
    body: `
      <h3>Intyget är baserat på</h3>
      <h4>Min undersökning av patienten</h4>
      <p>2023-02-07</p>
      <h4>Min telefonkontakt med patienten</h4>
      <p>Ej angivet</p>
      <h4>Journaluppgifter från den</h4>
      <p>Ej angivet</p>
      <h4>Ange vad annat är</h4>
      <p>Ej angivet</p>
    `,
  },
  {
    heading: 'Sysselsättning',
    body: `
      <h3>I relation till vilken sysselsättning bedömer du arbetsförmågan?</h3>
      <ul>
        <li>Nuvarande arbete</li>
      </ul>
      <h3>Ange yrke och arbetsuppgifter</h3>
      <p>Kontorsarbetare</p>

      <table>
        <thead>
          <tr>
            <th>Diagnoskod enligt ICD-10 SE</th>
            <th>Diagnos</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>S829</td>
            <td>Underbensfraktur med ospecificerad lokalisation</td>
          </tr>
        </tbody>
      </table>
    `,
  },
  {
    heading: 'Sjukdomens konsekvenser',
    body: `
      <h3>Funktionsnedsättning</h3>
      <h4>Beskriv undersökningsfynd, testresultat och observationer</h4>
      <p>Patienten har en underbensfraktur. Behandlas med helbensgrips och har därför nedsatt rörlighet i benet. Patienten
        får inte stödja på det skadade benet. Går långsamt och ostadigt med kryckor. Påtalar smärta vid undersökningen.
      </p>
      <h3>Aktivitetsbegränsning</h3>
      <h4>Beskriv vad patienten inte kan göra på grund av sin sjukdom. Ange vad uppgiften grundas
        på.</h4>
      <p>Patienten kan inte gå utan hjälpmedel. Klarar inte av att köra bil. Således mycket svårt att ta sig till arbetet.
        Dessutom problem med trappor på arbetsplatsen. Patienten kan även ha svårt att fokusera på sina arbetsuppgifter
        pga smärta.</p>
    `,
  },
  {
    heading: 'Medicinsk behandling',
    body: `
      <h3>Pågående medicinska behandlingar/åtgärder</h3>
      <h4>Ange vad syftet är och om möjligt tidplan samt ansvarig vårdenhet</h4>
      <p>Erhållit helbensgips, 0-belastning med kryckor. Smärtlindring.</p>
      <h3>Planerade medicinska behandlingar/åtgärder</h3>
      <h4>Ange vad syftet är och om möjligt tidplan samt ansvarig vårdenhet</h4>
      <p>Gips totalt 4 månader, initialt helbensgips. Rtg-kontroll efter 2, 4 och 6 veckor av frakturens läge. Om stabilt
        frakturläge vid 6 v kontroll byte till underbensgips. Rehabilitering efter avslutad gipsbehandling.</p>
    `,
  },
  {
    heading: 'Bedömning',
    body: `
      <h3>Min bedömning av patientens nedsättning av arbetsförmågan</h3>
      <table>
        <thead>
          <tr>
            <th>Nedsättningsgrad</th>
            <th>Från och med</th>
            <th>Till och med</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>100 procent</td>
            <td>2023-02-07</td>
            <td>2023-02-20</td>
          </tr>
        </tbody>
      </table>

      <h3>Patientens arbetsförmåga bedöms nedsatt längre tid än den som Socialstyrelsens
        försäkringsmedicinska beslutsstöd
        anger, därför att</h3>
      <p>Ej angivet</p>

      <h3>Kommer möjligheterna till återgång i arbete försämras om arbetstiden förläggs på annat
        sätt än att arbetstiden
        minskas lika mycket varje dag?</h3>
      <p>Ej angivet</p>

      <h3>Beskriv de medicinska skälen till att möjligheterna till återgång i arbete försämras</h3>
      <p>Ej angivet</p>

      <h3>Resor till och från arbetet med annat färdmedel än normalt kan göra det möjligt för
        patienten att återgå till
        arbetet under sjukskrivningsperioden.</h3>
      <p>Ej angivet</p>

      <h3>Prognos för arbetsförmåga utifrån aktuellt undersökningstillfälle</h3>
      <p>Patienten förväntas kunna återgå helt i nuvarande sysselsättning inom 1 månad.</p>
    `,
  },
  {
    heading: 'Åtgärder',
    body: `
      <h3>Här kan du ange åtgärder som du tror skulle göra det lättare för patienten att återgå i arbete</h3>
      <ul>
        <li>Arbetsanpassning</li>
      </ul>
      <h3>Här kan du ange fler åtgärder. Du kan också beskriva hur åtgärderna kan underlätta återgång i arbete.</h3>
      <p>Möjlighet till distansarbete i hemmet skulle underlätta mycket för patienten.</p>
    `,
  },
  {
    heading: 'Övriga upplysningar',
    body: `
      <h3>Övriga upplysningar</h3>
      <p>Ej angivet</p>
    `,
  },
  {
    heading: 'Kontakt',
    body: `
      <h3>Kontakt med Försäkringskassan</h3>
      <h4>Jag önskar att Försäkringskassan kontaktar mig</h4>
      <p>Ej angivet</p>
      <h4>Ange gärna varför du vill ha kontakt</h4>
      <p>Ej angivet</p>
    `,
  },
].map(({ heading, body }) => ({ heading, body: body.trim().replace(/\s+/g, ' ') }))
