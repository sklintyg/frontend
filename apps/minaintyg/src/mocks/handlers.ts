/* eslint-disable import/no-extraneous-dependencies */
import { fakeCertificateLabel, faker, fakerFromSchema } from '@frontend/fake'
import { rest } from 'msw'
import { certificateListItemSchema } from '../schema/certificateList.schema'
import { userSchema } from '../schema/user.schema'

export const handlers = [
  rest.get('/api/user', (_, res, ctx) =>
    res(
      ctx.status(200),
      ctx.json(
        fakerFromSchema(userSchema, {
          stringMap: {
            personName: faker.name.fullName,
          },
        })()
      )
    )
  ),
  rest.post('/api/certificates', (_, res, ctx) =>
    res(
      ctx.status(200),
      ctx.json({
        content: Array.from(
          { length: 5 },
          fakerFromSchema(certificateListItemSchema, {
            stringMap: {
              title: fakeCertificateLabel,
              certificateId: faker.datatype.uuid,
              timestamp: () => faker.date.between('2021-01-01T00:00:00.000Z', new Date().toISOString()).toISOString(),
            },
          })
        ),
      })
    )
  ),
]
