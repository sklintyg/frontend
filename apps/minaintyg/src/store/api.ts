import { getCookie } from '@frontend/utils'
import { isAnyOf, isPlainObject } from '@reduxjs/toolkit'
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
import { AvailableFunction, Certificate, CertificateMetadata } from '../schema/certificate.schema'
import { CertificateFilterOptions } from '../schema/certificateListFilter.schema'
import { User } from '../schema/user.schema'
import { CertificateFilterState } from './slice/certificateFilter.slice'

export const api = createApi({
  reducerPath: 'api',
  baseQuery: fetchBaseQuery({
    baseUrl: '/api/',
    prepareHeaders: (headers) => {
      if (getCookie('XSRF-TOKEN')) {
        headers.set('X-XSRF-TOKEN', getCookie('XSRF-TOKEN') ?? '')
      }
      return headers
    },
  }),
  tagTypes: ['User', 'Certificate'],
  endpoints: (builder) => ({
    getUser: builder.query<User, void>({
      query: () => 'user',
      providesTags: ['User'],
    }),
    getCertificates: builder.query<{ content: CertificateMetadata[] }, CertificateFilterState>({
      query: (body) => ({
        url: 'certificate',
        method: 'POST',
        body: Object.fromEntries(Object.entries(body).map(([key, value]) => [key, [value]])),
      }),
      providesTags: (result) =>
        result
          ? [...result.content.map(({ id }) => ({ type: 'Certificate' as const, id })), { type: 'Certificate', id: 'LIST' }]
          : [{ type: 'Certificate', id: 'LIST' }],
    }),
    getCertificatesFilter: builder.query<CertificateFilterOptions, void>({
      query: () => 'filters',
    }),
    getCertificate: builder.query<Certificate & { availableFunctions: AvailableFunction[] }, { id: string }>({
      query: ({ id }) => `certificate/${id}`,
      transformResponse: ({ certificate, availableFunctions }: { certificate: Certificate; availableFunctions: AvailableFunction[] }) => ({
        ...certificate,
        availableFunctions,
      }),
      providesTags: (result) => (result ? [{ type: 'Certificate' as const, id: result.metadata.id }] : []),
    }),
    sendCertificate: builder.mutation<void, { id: string }>({
      query: ({ id }) => ({
        url: `certificate/${id}/send`,
        method: 'POST',
      }),
      invalidatesTags: (result, error, { id }) => (error ? [] : [{ type: 'Certificate', id }]),
    }),
  }),
})

export const { useGetCertificatesQuery, useGetCertificateQuery, useGetCertificatesFilterQuery, useSendCertificateMutation } = api

export const isFulfilledEndpoint = isAnyOf(...Object.values(api.endpoints).map((endpoint) => endpoint.matchFulfilled))
export const isRejectedEndpoint = isAnyOf(...Object.values(api.endpoints).map((endpoint) => endpoint.matchRejected))
export const hasResponse = (o: unknown): o is { response: Response } =>
  isPlainObject(o) && 'response' in o && o.response instanceof Response
