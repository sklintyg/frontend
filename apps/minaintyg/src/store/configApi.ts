import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
import { AppConfig } from '../schema/appConfig.schema'

export const configApi = createApi({
  reducerPath: 'api',
  baseQuery: fetchBaseQuery({
    baseUrl: '/appconfig/api/',
  }),
  tagTypes: ['User'],
  endpoints: (builder) => ({
    getAppConfig: builder.query<AppConfig, void>({
      query: () => 'app',
    }),
    getCertificateDescription: builder.query<CertificateDescription, void>({
      query: () => 'map',
    }),
  }),
})

export const { useGetAppConfigQuery, useGetCertificateDescriptionQuery } = configApi
