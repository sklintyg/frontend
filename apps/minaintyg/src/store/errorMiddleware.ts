import { randomUUID } from '@frontend/utils'
import { isRejectedWithValue, Middleware, MiddlewareAPI, SerializedError } from '@reduxjs/toolkit'
import { FetchBaseQueryError } from '@reduxjs/toolkit/dist/query'

export type ErrorObject = (FetchBaseQueryError & { id?: string }) | (SerializedError & { id?: string })

/**
 * Error handling middleware
 * https://redux-toolkit.js.org/rtk-query/usage/error-handling
 */
export const errorMiddleware: Middleware = (_: MiddlewareAPI) => (next) => (action) => {
  if (isRejectedWithValue(action)) {
    const id = randomUUID()
    // const { method, url } = action.meta.baseQueryMeta.request
    // const { status } = action.meta.baseQueryMeta.response

    // Append identifier to payload
    return next(Object.assign(action, { payload: { ...action.payload, id } }))
  }

  return next(action)
}
