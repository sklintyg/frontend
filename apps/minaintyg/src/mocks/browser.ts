/* eslint-disable import/no-extraneous-dependencies */
import { faker } from 'fake'
import { setupWorker } from 'msw'
import { handlers } from './handlers'

faker.setLocale('sv')

export const worker = setupWorker(...handlers)
