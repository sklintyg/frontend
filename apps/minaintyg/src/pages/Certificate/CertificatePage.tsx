import { IDSSpinner } from '@frontend/ids-react-ts'
import { skipToken } from '@reduxjs/toolkit/query'
import { useParams } from 'react-router-dom'
import { PageDivider } from '../../components/PageDivider/PageDivider'
import { PageHeading } from '../../components/PageHeading/PageHeading'
import { PageHeadingDescription } from '../../components/PageHeading/PageHeadingDescription'
import { useGetCertificateQuery } from '../../store/api'
import { CertificateActions } from './components/CertificateActions/CertificateActions'
import { CertificateBody } from './components/CertificateBody/CertificateBody'
import { CertificateEventsInfo } from './components/CertificateEventsInfo/CertificateEventsInfo'
import { CertificateFooter } from './components/CertificateFooter'
import { CertificateInformation } from './components/CertificateInformation'
import { CertificateReplacedAlert } from './components/CertificateReplacedAlert'
import { CertificateStatusBadge } from './components/CertificateStatusBadge'
import { ReadCertificateError } from './components/ReadCertificateError'
import { CertificateAttentionAlert } from './components/CertificateAttentionAlert/CertificateAttentionAlert'
import { CertificatePreambleText } from './components/CertificatePremableText/CertificatePreambleText'
import { ReadMoreAboutAction } from './components/ReadMoreAboutDialog/ReadMoreAboutAction'

const FALLBACK_DESCRIPTION = `Det här är ditt intyg. Intyget innehåller all information vården fyllt i. Du kan inte ändra något i ditt intyg. Har du frågor kontaktar du den som skrivit ditt intyg.`

export function CertificatePage() {
  const { id } = useParams()
  const { data: certificate, isLoading, error } = useGetCertificateQuery(id ? { id } : skipToken)

  return (
    <>
      <PageHeading heading="Läs och hantera ditt intyg">
        {certificate && (
          <>
            <CertificatePreambleText texts={certificate.texts} />
            {certificate.metadata.statuses.includes('REPLACED') && <CertificateReplacedAlert />}
          </>
        )}
        {error && <PageHeadingDescription>{FALLBACK_DESCRIPTION}</PageHeadingDescription>}
        <ReadMoreAboutAction />
      </PageHeading>

      {isLoading && <IDSSpinner data-testid="spinner" />}
      {error && <ReadCertificateError id={id} error={error} />}
      {certificate && (
        <>
          <div className="mb-5">
            <CertificateAttentionAlert availableFunctions={certificate.availableFunctions} />
            <CertificateActions
              recipient={certificate.metadata.recipient}
              availableFunctions={certificate.availableFunctions}
              id={certificate.metadata.id}
            />
          </div>

          <div className="md:hidden">
            <PageDivider />
          </div>

          <div className="rounded-[10px] border-stone-line md:border md:p-7 md:shadow-[0_0_2px_0px_rgba(0,0,0,0.3)]">
            <div className="mb-5 flex flex-col justify-between gap-2.5 md:flex-row md:gap-5">
              <h2 className="ids-heading-2">{certificate.metadata.type.name}</h2>
              <div className="flex gap-1">
                {certificate.metadata.statuses.map((status) => (
                  <CertificateStatusBadge key={status} status={status} />
                ))}
              </div>
            </div>
            <div className="mb-5">
              <CertificateInformation {...certificate.metadata} />
            </div>
            <h3 className="ids-heading-4 mb-0">Händelser</h3>
            <CertificateEventsInfo events={certificate.metadata.events} />

            <article className="ids-certificate">
              <CertificateBody content={certificate.content} />
              <CertificateFooter {...certificate.metadata} />
            </article>
          </div>
        </>
      )}
    </>
  )
}
