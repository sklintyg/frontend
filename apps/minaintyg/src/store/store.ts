import { configureStore } from '@reduxjs/toolkit'
import { api } from './api'
import { errorMiddleware } from './middleware/error.middleware'
import { reducer } from './reducer'
import { idpMiddleware } from './slice/idpMiddleware'

export const store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat([api.middleware, idpMiddleware, errorMiddleware]),
})

export type AppDispatch = typeof store.dispatch
