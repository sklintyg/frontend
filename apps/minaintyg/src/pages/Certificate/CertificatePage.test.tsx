import { fakerFromSchema } from '@frontend/fake'
import { render, screen, waitFor } from '@testing-library/react'
import { rest } from 'msw'
import { Provider } from 'react-redux'
import { createMemoryRouter, createRoutesFromChildren, Route, RouterProvider } from 'react-router-dom'
import { server } from '../../mocks/server'
import {
  AvailableFunctionsTypeEnum,
  CertificateMetadata,
  certificateMetadataSchema,
  certificateRecipientSchema,
  certificateSchema,
} from '../../schema/certificate.schema'
import { store } from '../../store/store'
import { CertificatePage } from './CertificatePage'

function renderComponent(metadata: CertificateMetadata) {
  server.use(
    rest.get('/api/certificate/:id', (_, res, ctx) =>
      res(
        ctx.json({
          certificate: fakerFromSchema(certificateSchema)({
            metadata,
            content: [{ heading: 'Rubrik 1', body: '<p>Test</p>' }],
          }),
          availableFunctions: [
            {
              type: AvailableFunctionsTypeEnum.enum.SEND_CERTIFICATE,
            },
          ],
        })
      )
    )
  )

  return render(
    <Provider store={store}>
      <RouterProvider
        router={createMemoryRouter(createRoutesFromChildren([<Route key="root" path="/:id" element={<CertificatePage />} />]), {
          initialEntries: ['/12345'],
        })}
      />
    </Provider>
  )
}

it('Should have article content', async () => {
  renderComponent(
    fakerFromSchema(certificateMetadataSchema)({
      id: '0b295069-d03c-413c-9f4a-b6fd81ad2d87',
      issuer: {
        name: 'Ajla Doktor',
      },
      unit: {
        name: 'Stockholm',
        address: 'Vägen 12',
        zipCode: '123 12',
        city: 'Söder',
        phoneNumber: '123 112 1212',
      },
      careUnit: {
        name: 'Stockholm',
        address: 'Vägen 12',
        zipCode: '123 12',
        city: 'Söder',
        phoneNumber: '123 112 1212',
      },
      issued: '2023-10-06T15:20:28',
      recipient: fakerFromSchema(certificateRecipientSchema)({
        sent: null,
      }),
      events: [],
    })
  )
  expect(await screen.findByTestId('spinner')).toBeInTheDocument()
  expect(await screen.findByRole('article')).toBeInTheDocument()
  expect(screen.getByRole('article')).toMatchSnapshot()
})

it('Should display alert message when certificate is replaced', async () => {
  renderComponent(
    fakerFromSchema(certificateMetadataSchema)({
      recipient: null,
      issued: '2023-10-06T15:20:28',
      statuses: ['REPLACED'],
    })
  )

  expect(await screen.findByRole('alert')).toBeInTheDocument()
  expect(screen.getByRole('alert')).toMatchInlineSnapshot(`
    <ids-alert
      role="alert"
    >
      Läkaren kan ersätta ett intyg om till exempel intyget innehåller fel information eller om ny information tillkommit.
    </ids-alert>
  `)
})

describe('Unable to load certificate', () => {
  function renderWithFault() {
    server.use(rest.get('/api/certificate/:id', (_, res, ctx) => res(ctx.status(500))))

    render(
      <Provider store={store}>
        <RouterProvider
          router={createMemoryRouter(createRoutesFromChildren([<Route key="root" path="/:id" element={<CertificatePage />} />]), {
            initialEntries: ['/12345'],
          })}
        />
      </Provider>
    )
  }

  it('Should render error message', async () => {
    renderWithFault()
    await waitFor(() => expect(screen.queryByTestId('spinner')).not.toBeInTheDocument())
    expect(screen.getAllByRole('alert')).toMatchSnapshot()
  })

  it('Should display fallback description', async () => {
    renderWithFault()
    await waitFor(() => expect(screen.queryByTestId('spinner')).not.toBeInTheDocument())
    expect(screen.getByText(/det här är ditt intyg/i)).toBeInTheDocument()
    expect(screen.getByText(/det här är ditt intyg/i)).toHaveClass('ids-preamble')
  })
})
