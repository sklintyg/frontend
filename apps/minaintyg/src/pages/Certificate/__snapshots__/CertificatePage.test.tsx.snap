// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should have article content 1`] = `
<article
  class="ids-certificate"
>
  <div
    class="border-b border-neutral-90"
  >
    <div
      class="ids-accordion ids-accordion--lean"
    >
      <div
        aria-controls=":r6:"
        aria-expanded="false"
        class="ids-accordion__button ids-accordion__button--m"
        role="button"
        tabindex="0"
      >
        <div
          class="ids-accordion__headline ids-accordion__headline-1 ids-accordion__headline--m"
        >
          <h2>
            Rubrik 1
          </h2>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="ids-accordion__content"
        id=":r6:"
      >
        <p>
          Test
        </p>
      </div>
    </div>
  </div>
  <footer
    class="pt-7"
  >
    <h2
      class="ids-heading-s !normal-case"
    >
      Intyget är skrivet av
    </h2>
    <address
      class="flex flex-col gap-4 not-italic md:flex-row md:gap-[6.25rem]"
    >
      <div>
        <strong>
          Ajla Doktor
        </strong>
        <p>
          123 112 1212
        </p>
      </div>
      <div>
        <strong>
          Stockholm
        </strong>
        <p>
          Vägen 12, 123 12 Söder
        </p>
      </div>
    </address>
    <hr
      class="my-7 border-neutral-90"
    />
    <strong>
      Intygs-ID:
    </strong>
     
    0b295069-d03c-413c-9f4a-b6fd81ad2d87
  </footer>
</article>
`;

exports[`Unable to load certificate > Should render error message 1`] = `
[
  <div
    class="ids-alert ids-alert--error"
    role="alert"
  >
    <div
      class="ids-alert__header"
    >
      <div
        class="ids-alert__icon_and_text"
      >
        <span
          class="ids-alert__state-icon"
          title=""
        />
        <div
          class="ids-alert__headline"
        >
          <h2>
            Tekniskt fel
          </h2>
        </div>
      </div>
    </div>
    <div
      class="ids-alert__content"
    >
      <div
        class="flex flex-col gap-4"
      >
        <p>
          Intyget kunde inte visas på grund av ett tekniskt fel. Försök igen senare. Om felet kvarstår kontakta 
          <a
            class="ids-link ids-link--underlined"
            href="https://www.1177.se/e-tjanster-support"
            rel="noreferrer"
            target="_blank"
          >
            support
            <span
              class="ids-icon-external-link-small  ids-icon--text-end"
            />
          </a>
          och ange nedan fel-id:
        </p>
        <div
          class=""
        >
          <button
            aria-label="Kopiera fel-id"
            tabindex="0"
            type="button"
            value="5f92e947-e2ee-4238-bf29-4cdc6b6c4b54"
          >
            <p
              class="flex justify-center gap-1 align-bottom font-bold"
              data-state="closed"
            >
              <span>
                Fel-id: 
              </span>
               
              <span
                class="font-normal"
              >
                5f92e947-e2ee-4238-bf29-4cdc6b6c4b54
              </span>
              <ids-icon-copy-file
                class="ids-icon"
                size="s"
              />
            </p>
            <p
              class="flex justify-items-start"
            />
          </button>
        </div>
        <p>
          Om du inte kan nå ditt intyg, kontakta din mottagning för att få en kopia av intyget.
        </p>
      </div>
    </div>
  </div>,
]
`;
